src/citation/utils.py

```diff
@@ -1,34 +1,87 @@
 from urllib.parse import urlparse
 
-import pdf2doi
+import requests
 from django.contrib.postgres.search import SearchQuery
+from django.core.files.storage import default_storage
 from django.db.models import Q
+from django.http.request import HttpRequest
+from rest_framework.request import Request
 
-from citation.constants import CITATION_TYPE_FIELDS, JOURNAL_ARTICLE
+from citation.constants import JOURNAL_ARTICLE
+from citation.exceptions import GrobidProcessingError
 from citation.models import CitationEntry
-from citation.schema import generate_json_for_journal
+from citation.schema import generate_json_for_doi, generate_json_for_pdf
+from citation.serializers import CitationEntrySerializer
 from paper.models import Paper
 from paper.paper_upload_tasks import celery_process_paper
 from paper.serializers import PaperSubmissionSerializer
+from researchhub.settings import AWS_STORAGE_BUCKET_NAME, GROBID_SERVER
+from user.models import User
+from utils.sentry import log_error
 
 
-def get_citation_entry_from_pdf(pdf, user_id, organization_id, project_id):
-    conversion = pdf2doi.pdf2doi_singlefile(pdf)
-    json = generate_json_for_journal(conversion)
+def get_pdf_header_data(path):
+    url = f"{GROBID_SERVER}/header_extract"
+    request_data = {
+        "s3_file_path": f"{path}",
+        "bucket_name": f"{AWS_STORAGE_BUCKET_NAME}",
+    }
+    try:
+        response = requests.post(url, data=request_data, timeout=10)
+        data = response.json()
+        status = data.get("status")
+        if status == 200:
+            return data.get("data", {})
+    except requests.ConnectionError as e:
+        log_error(e)
```

Comment:
could you add a suffix or beginning - phrases here so that we can easily identify what the error was for when we look in sentry without reading through the errors at first glance?

Reply:
I'll make that change